lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -225,125 +325,172 @@ def create_launch_config(connection, module):
-    if launch_configs[0].block_device_mappings is not None:
-        result['block_device_mappings'] = []
-        for bdm in launch_configs[0].block_device_mappings:
-            result['block_device_mappings'].append(dict(device_name=bdm.device_name, virtual_name=bdm.virtual_name))
-            if bdm.ebs is not None:
-                result['block_device_mappings'][-1]['ebs'] = dict(snapshot_id=bdm.ebs.snapshot_id, volume_size=bdm.ebs.volume_size)
+            result['InstanceMonitoring'] = False
+    if launch_config.get('BlockDeviceMappings') is not None:
+        result['BlockDeviceMappings'] = []
+        for bdm in launch_config.get('BlockDeviceMappings'):
+            result['BlockDeviceMappings'].append(dict(device_name=bdm.get(
+                'DeviceName'), virtual_name=bdm.get('VirtualName')))
+            if bdm.get('Ebs') is not None:
+                result['BlockDeviceMappings'][-1]['ebs'] = dict(
+                    snapshot_id=bdm.get('Ebs').get('SnapshotId'),
+                    volume_size=bdm.get('Ebs').get('VolumeSize'))
 
     if user_data_path:
-        result['user_data'] = "hidden" # Otherwise, we dump binary to the user's terminal
+        result['UserData'] = "hidden"
+
+    return_object = {
+        'Name': result.get('LaunchConfigurationName'),
+        'created_time': result.get('CreatedTime'),
+        'ImageId': result.get('ImageId'),
+        'Arn': result.get('LaunchConfigurationARN'),
+        'SecurityGroups': result.get('SecurityGroups'),
+        'InstanceType': result.get('InstanceType'),
+        'Result': result
+    }
 
-    module.exit_json(changed=changed, name=result['name'], created_time=result['created_time'],
-                     image_id=result['image_id'], arn=result['launch_configuration_arn'],
-                     security_groups=result['security_groups'],
-                     instance_type=result['instance_type'],
-                     result=result)
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(return_object))
 
 
 def delete_launch_config(connection, module):
-    name = module.params.get('name')
-    launch_configs = connection.get_all_launch_configurations(names=[name])
-    if launch_configs:
-        launch_configs[0].delete()
-        module.exit_json(changed=True)
-    else:
-        module.exit_json(changed=False)
+    try:
+        name = module.params.get('name')
+        launch_configs = connection.describe_launch_configurations(
+            LaunchConfigurationNames=[name]).get('LaunchConfigurations')
+        if launch_configs and len(launch_configs) > 0:
+            connection.delete_launch_configuration(
+                LaunchConfigurationName=launch_configs[0].get(
+                    'LaunchConfigurationName'))
+            module.exit_json(changed=True)
+        else:
+            module.exit_json(changed=False)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
```

Comment:
Same as the comment for the other ClientError handling. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2